### ISSUE TITLE
4.1 - Debian netvm does not provide network

### ISSUE THREAD

**Issue by unman** [QubesOS/qubes-issues#6696]
**Qubes OS version**

4.1 with testing repositories enabled

**Are you using any [testing](https://www.qubes-os.org/doc/testing/) repositories? If so, please list them and where they are enabled (i.e., dom0, TemplateVM, or StandaloneVM).**

dom0, Debian-10

**Are you providing feedback about a specific package or packages in testing? (If so, please list the package(s) here and [provide feedback in `updates-status`](https://www.qubes-os.org/doc/testing/#providing-feedback), linking to this issue.)**

**Affected component(s) or functionality**

Networking

**Brief summary**

Using Debian 10, and standard configuration, sys-net does not provide network to attached qubes

**How Reproducible**

100%

**To Reproduce**

1. Install 4.1 with Debian default 
2. 
3. 

**Expected behavior**

sys-net will provide network

**Actual behavior**

vifs are created for attached qubes, but they are not active and no IP address is configured.
As result, network in sys-net is working, but attached qubes are offline.

**Screenshots**

**Additional context**

**Solutions you've tried**

**Relevant [documentation](https://www.qubes-os.org/doc/) you've consulted**

**Related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues**

**Comment by GWeck** [comment]
This seems to depend on the avaiable network interfaces. On my machine (an HP Elitebook 840 G4), I get network connectivity even with debian (default) sys-net, but only via ethernet. The WiFi interface is not shown and so I cannot confgure it at all.

If I switch sys-net to fedora-33, both ethernet and WiFi are working.

**Comment by andrewdavidwong** [comment]
> Are you using any testing repositories? If so, please list them and where they are enabled (i.e., dom0, TemplateVM, or StandaloneVM).
> 
> dom0, Debian-10

FYI, this states the locations but not which testing repos are enabled.

**Comment by unman** [comment]
On Mon, Jun 14, 2021 at 04:25:33AM -0700, Dr. Gerhard Weck wrote:
> This seems to depend on the avaiable network interfaces. On my machine (an HP Elitebook 840 G4), I get network connectivity even with debian (default) sys-net, but only via ethernet. The WiFi interface is not shown and so I cannot confgure it at all.
> 
> If I switch sys-net to fedora-33, both ethernet and WiFi are working.
> 

This is likely a driver issue in the Debian qube.
On my machines the WiFi interface is up and working, but the vif
interfaces are not configured.
I guessed that this was not a Fedora issue.
I will test it with a connected ethernet cable and update.

**Comment by GWeck** [comment]
After installing the driver package `iwlwifi` for Intel WiFi firmware, network access works in a debian based sys-net via WiFi, too, in Qubes R4.0.4 and R4.1. AppVMs using this sys-net VM have no problems at all. So, for me, this were debian driver problems, after all.

**Comment by unman** [comment]
On Wed, Jun 16, 2021 at 04:43:29AM -0700, Dr. Gerhard Weck wrote:
> After installing the driver package `iwlwifi` for Intel WiFi firmware, network access works in a debian based sys-net via WiFi, too, in Qubes R4.0.4 and R4.1. AppVMs using this sys-net VM have no problems at all. So, for me, this were debian driver problems, after all.
> 

I elected to *only* use Debian-10.
I'll test with a new release and installing both templates, and
using only Debian.

**Comment by a-barinov** [comment]
I can confirm the problem of non-initialising vif interfaces existed in May and June builds of 4.1. In the 4.1-beta1 the problem is gone.

**Comment by unman** [comment]
Confirmed @a-barinov findings - issue resolved in 4.1-beta1

**Comment by GWeck** [comment]
debian-11 is a new release, so it is probably lacking the `iwlwifi` package, like was the case for debian-10. Could you try to install this package in the debian-11 template, and then check again?

**Comment by resulin** [comment]
> IIRC, `iwlwifi` was loaded according to `lsmod`. Unfortunately, I can't reinstall to test again.

I can confirm, here's the output from running `lsmod` on a `debian-11` based `sys-net`:

```
Module                  Size  Used by
xt_nat                 16384  4
xen_netback            77824  0
nft_ct                 20480  2
xt_REDIRECT            20480  1
ipt_REJECT             16384  1
nf_reject_ipv4         16384  1 ipt_REJECT
xt_state               16384  0
joydev                 28672  0
xt_conntrack           16384  4
nft_counter            16384  30
nft_chain_nat          16384  3
xt_MASQUERADE          20480  1
nf_nat                 57344  4 xt_nat,nft_chain_nat,xt_MASQUERADE,xt_REDIRECT
nf_conntrack          167936  7 xt_conntrack,nf_nat,nft_ct,xt_state,xt_nat,xt_MASQUERADE,xt_REDIRECT
nf_defrag_ipv6         24576  1 nf_conntrack
nf_defrag_ipv4         16384  1 nf_conntrack
nft_compat             20480  18
nf_tables             262144  89 nft_ct,nft_compat,nft_counter,nft_chain_nat
nfnetlink              20480  2 nft_compat,nf_tables
intel_rapl_msr         20480  0
intel_rapl_common      32768  1 intel_rapl_msr
crct10dif_pclmul       16384  1
crc32_pclmul           16384  0
crc32c_intel           24576  0
ghash_clmulni_intel    16384  0
iwlwifi               364544  0
drm_vram_helper        24576  0
snd_pcm               135168  0
drm_ttm_helper         16384  1 drm_vram_helper
ttm                   118784  2 drm_vram_helper,drm_ttm_helper
cfg80211              995328  1 iwlwifi
snd_timer              49152  1 snd_pcm
drm_kms_helper        282624  1 drm_vram_helper
snd                   110592  2 snd_timer,snd_pcm
soundcore              16384  1 snd
pcspkr                 16384  0
e1000e                315392  0
serio_raw              20480  0
cec                    61440  1 drm_kms_helper
rfkill                 32768  2 cfg80211
ehci_pci               20480  0
ehci_hcd              110592  1 ehci_pci
floppy                 94208  0
i2c_piix4              28672  0
ata_generic            16384  0
pata_acpi              16384  0
xen_privcmd            28672  54
xen_gntdev             45056  2
xen_gntalloc           20480  6
xen_blkback            53248  0
xen_evtchn             16384  7
parport_pc             40960  0
ppdev                  24576  0
drm                   643072  4 drm_kms_helper,drm_vram_helper,drm_ttm_helper,ttm
lp                     20480  0
parport                73728  3 parport_pc,lp,ppdev
fuse                  176128  3
bpf_preload            16384  0
ip_tables              36864  0
overlay               151552  1
xen_blkfront           57344  10
```

Installing the package `firmware-iwlwifi` (which is not installed by default) fixes the issue.

**Comment by resulin** [comment]
The issue is due to the `debian-11` template lacking the `firmware-iwlwifi` package. You can label it as diagnosed again @andrewdavidwong

**Comment by unman** [comment]
This has **nothing** to do with the absence of firmware or drivers.
My original report made it clear that networking was working in sys-net - it follows that any needed firmware was installed.
What was *not* working was networking down to vif.

The issue ws resolved in a later build of 4.1.
I am therefore closing this issue.
